### PR TITLE
[SPARK-37498][PYTHON] Add eventually for test_reuse_worker_of_parallelize_range

### PR DESCRIPTION
**Pull request by Yikun**
### What changes were proposed in this pull request?
Add eventually for test_reuse_worker_of_parallelize_range

### Why are the changes needed?
Avoid test_reuse_worker_of_parallelize_range becoming flaky when resources are tight or some other reason

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT passed.

### CODE REVIEW

File: python/pyspark/tests/test_worker.py [hunk 1]
```diff
@@ -191,8 +191,14 @@ def test_reuse_worker_of_parallelize_range(self):
         rdd = self.sc.parallelize(range(20), 8)
```
Comment (HyukjinKwon) [inline python/pyspark/tests/test_worker.py:191]: @Yikun, maybe just run the eventually with the whole test here. e.g.)

```python
def test_reuse_worker_of_parallelize_range(self):
    def check_reuse_worker_of_parallelize_range():
        rdd = self.sc.parallelize(range(20), 8)
        previous_pids = rdd.map(lambda x: os.getpid()).collect()
        current_pids = rdd.map(lambda x: os.getpid()).collect()

        for pid in current_pids:
            self.assertTrue(pid in previous_pids)

    eventually(check_reuse_worker_of_parallelize_range, catch_assertions=True)
```

Comment (Yikun) [reply python/pyspark/tests/test_worker.py]: Ah, yes for sure

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by HyukjinKwon** [comment]
Seems like this happens in branch-3.2 too. Let me backport this. Thanks for letting me know @huaxingao

**Comment by HyukjinKwon** [comment]
Merged to branch-3.2 too.

**Pull request closed without merging**